_posts/2023-08-21-Debezium-UI-connectors-custom-property.adoc

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title:  Enable pass-through of additional properties in Debezium UI
+date:   2023-08-21
+tags: [ community, debezium-ui ]
+author: indrashukla
+---
+The Debezium UI team continues to add support for more features, allowing users to more easily configure connectors. In this article, we'll describe and demonstrate the how to provide the additional properties that are not specified in the previous sections of connector configuration. 
+Read further for more information!
+
++++<!-- more -->+++
+
+== Need for custom properties step
+
+Debezium is very flexible - each connector can be configured and fine-tuned in a variety of ways. Debezium UI already expose an extensive list of properties that are segregated under different steps for configuration. But even with this extensive list of exposed properties in UI there are few properties that are not know prior and are need to configure while connector setup.
+
+*For example*
+
+Let’s take an example of oracle connector.
```

Comment:
@Naros @rk3rn3r I am looking for example properties we may provide as additional properties here for a connector. And some bit of reasoning like why we don't have prior knowledge and could not be exposed via UI

Reply:
I think all `producer.*` and `consumer.*` properties are additional options, as well as all pass-through options for the database drivers starting with `database.` and that are not part of the UI properties / the documented schema.

Reply:
Hi @rk3rn3r and @indraraj, just one reminder that we changed the `database.` prefix to `driver.` prefix for passthru driver properties in Debezium 2.0.  The only properties that begin with `database.` should generally focus only on values used by Debezium proper.

Oracle properties can be found at:
https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/jdbc/OracleConnection.html

Some example driver properties that might be useful for Oracle `oracle.net.keepAlive` which is useful when deploying on AWS with GLB enabled, `oracle.net.disableOob` if the OOB isn't disabled at the TNS listener level, as well as various timeouts managed by the driver.